src/main/scala/org/broadinstitute/hail/Utils.scala

```diff
@@ -407,6 +408,21 @@ class RichRDDByteArray(val r: RDD[Array[Byte]]) extends AnyVal {
   }
 }
 
+class RichPairRDD[K, V](val r: RDD[(K, V)])
+                       (implicit kt: ClassTag[K], vt: ClassTag[V], ord: Ordering[K] = null) {
+
+  def leftOuterJoinDistinct[W](other: RDD[(K, W)]): RDD[(K, (V, Option[W]))] = leftOuterJoinDistinct(other, defaultPartitioner(r, other))
+
+  def leftOuterJoinDistinct[W](other: RDD[(K, W)], partitioner: Partitioner) = {
```

Comment:
> the left RDD values are always unique in our current schema (va, gs).

So unfortunately, that's not true as things currently stand.  I asked for clarification on this, and VCF does not forbid lines with duplicate variants (ugh).  So for now, all operations should preserve the collection of variants, including number, even when there are duplicates.

Now I think this is what we want:

```
  def leftOuterJoinDistinct[W](other: RDD[(K, Option[W])], partitioner: Partitioner) = {
    r.cogroup(other, partitioner).flatMapValues { pair =>
      val w = pair._2.headOption
      pair._1.map((_, w))
    }
  }
```

This will join other with r as if the keys of other were unique.  It will preserve the collection keys in r, and therefore should be safe to use everywhere.

It is necessary to use this in filtervariants list since without it duplicate variants in the list will lead to duplicate variants in the output.

I notice that the `keep` case of filtervariants list is also wrong.  We additionally need a joinDistinct that doesn't duplicate variants if there are duplicates in the variant list.

Make sense?